### DISCUSSION TITLE
Raw Query with $executeRaw Fails When Using Variable in SQL CREATE SEQUENCE

### DISCUSSION THREAD

**Discussion by wangyi12358** [prisma/prisma#24820] [Q&A] [answered]
### Bug description

When using the Prisma $executeRaw method to create a SQL sequence with a variable for the sequence name, the query fails with a syntax error. The error message indicates that the variable is not being properly substituted into the SQL query.

This is my code:
```js
const name = 'tenant_code_seq';
await prisma.$executeRaw`CREATE SEQUENCE ${name} START 1`;
```

The query fails with the following error message:
```bash
Raw query failed. Code: `42601`. Message: `ERROR: syntax error at or near "$1"`
```

Debug info: 
![image](https://github.com/user-attachments/assets/2ed64191-c737-4e1b-a297-eaf24b4fd6f5)

But if I don't use variable names to execute, there won't be any errors.
```js
await this.$executeRaw`CREATE SEQUENCE tenant_code_seq START 1`;
```

Using this method will not result in any errors.
```js
const sql = `CREATE SEQUENCE ${name} START 1`;
await this.$executeRawUnsafe(sql);
```
I am currently using this method to temporarily solve the problem.

### How to reproduce

1. Use Prisma Client to execute a raw SQL query with $executeRaw to create a sequence.
2. Use a variable to specify the sequence name.
```js
const name = 'tenant_code_seq';
await prisma.$executeRaw`CREATE SEQUENCE ${name} START 1`;
```
3. Run the code.

### Expected behavior

The sequence should be created successfully with the specified name.

### Prisma information

```prisma
// Add your schema.prisma
```

```ts
async function checkAndCreateSequence(name: string) {
  const exists = await this.$queryRaw`
    SELECT EXISTS (
      SELECT 1
      FROM pg_class
      WHERE relkind = 'S' AND relname = ${name}
    ) as exists
  `.then((res) => res[0].exists);
  if (!exists) {
    await this.$executeRaw`CREATE SEQUENCE ${name} START 1`;
  }
}
```

### Environment & setup

- OS: MacOS M2
- Database: PostgreSQL 14.11
- Node.js version: 20.11.1

### Prisma Version

```
5.14.0
```

**Comment by SevInf** [comment, marked as answer]
You can not use `$executeRaw` for these purpose and `$executeRawUnsafe` is the right way to go in this case: "safe" version creates [prepared statement](https://www.postgresql.org/docs/current/sql-prepare.html) and those support parameter substitution only in certain positions that do not include column or table names.

**Reply by wangyi12358** [reply]
OK thank you so much
